src/gather/tests/test_commands.py

```diff
@@ -42,14 +59,42 @@ def test_collecting(self):
     def test_non_collision(self):
         nice = NICE_COMMANDS.collect()
         evil = EVIL_COMMANDS.collect()
-        self.assertIs(nice['foo'], main2)
+        self.assertIs(nice['weird_name'], main2)
         self.assertIs(nice['bar'], main3)
-        self.assertIs(evil['foo'], main3)
+        self.assertIs(evil['weird_name'], main3)
 
     def test_cross_module_collection(self):
         collected = _helper.WEIRD_COMMANDS.collect()
         self.assertIn('weird_function', collected)
 
+    def test_one_of_strategy(self):
+        collected = CONFLICTING_COMMANDS.collect()
+        weird_name = collected.pop('weird_name')
+        self.assertEquals(collected, {})
+        self.assertIn(weird_name, (weird_name1, weird_name2, weird_name3))
```

Comment:
Re: docstring for test methods -- none of them have them, and I set pylint to disable warnings on missing docstrings. I'm going to open a separate issue to enable pylint docstring enforcement (which, by definition, will lead to docstrings).